azurerm/resource_arm_automation_schedule.go

```diff
@@ -0,0 +1,184 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/automation"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationScheduleCreateUpdate,
+		Read:   resourceArmAutomationScheduleRead,
+		Update: resourceArmAutomationScheduleCreateUpdate,
+		Delete: resourceArmAutomationScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"start_time": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
could we add some validation to check this is at least 5 minutes in the future? we can do that with a `ValidateFunc` in the Schema. Submitting a date in the past returns the following error which isn't particularly helpful:

```
* azurerm_automation_schedule.example: automation.ScheduleClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Unknown" Message="Unknown service error"
```

Reply:
Done